presto-hive/src/main/java/com/facebook/presto/hive/ColumnarTextHiveRecordCursor.java

```diff
@@ -170,64 +174,65 @@ public ColumnarTextHiveRecordCursor(
             types[i] = typeManager.getType(column.getTypeSignature());
             hiveTypes[i] = column.getHiveType();
 
-            if (!column.isPartitionKey()) {
+            if (!column.isPartitionKey() && !column.isHidden()) {
```

Comment:
You forgot to change `parseXxxColumns` in this file. Double check and see if there are other places where you missed the same thing.